### PR TITLE
Enable SPI on STM32WB

### PR DESCRIPTION
**Pull request by rickconrey**
This contribution enables SPI1 and SPI2 on the STM32WB

### CODE REVIEW

**Comment by erwango** [comment]
Please rebase

File: boards/arm/nucleo_wb55rg/nucleo_wb55rg.dts [hunk 1]
```diff
@@ -77,6 +77,10 @@ arduino_i2c: &i2c1 {
 	status = "okay";
 };
 
+&spi1 {
```
Comment (erwango) [inline boards/arm/nucleo_wb55rg/nucleo_wb55rg.dts]: Can you add it as arduino_spi? (cf arduino_serial)

File: tests/drivers/spi/spi_loopback/boards/nucleo_wb55rg.conf [hunk 2]
```diff
@@ -0,0 +1,2 @@
+CONFIG_SPI_LOOPBACK_DRV_NAME="SPI_1"
+CONFIG_SPI_1=y
```
Comment (erwango) [inline tests/drivers/spi/spi_loopback/boards/nucleo_wb55rg.conf]: This last line should not be needed

**Review by erwango — CHANGES_REQUESTED** [review]
Please also update board documentation and board's Kconfig.defconfig.

File: boards/arm/nucleo_wb55rg/nucleo_wb55rg.dts (hunk 1 above)
Comment (rickconrey) [reply boards/arm/nucleo_wb55rg/nucleo_wb55rg.dts]: changed

File: tests/drivers/spi/spi_loopback/boards/nucleo_wb55rg.conf (hunk 2 above)
Comment (rickconrey) [reply tests/drivers/spi/spi_loopback/boards/nucleo_wb55rg.conf]: removed!

**Review by erwango — APPROVED** [review]

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst [hunk 3]
```diff
@@ -190,6 +192,10 @@ Default Zephyr Peripheral Mapping:
 - LD1 : PB5
 - LD2 : PB0
 - LD3 : PB1
+- SPI_1_NSS : PA4 (arduino_spi)
```
Comment (dbkinder) [inline boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: The commit message says, "enables SPI1 and SPI2", but only looks like SPI1 is documented?

**Review by dbkinder — CHANGES_REQUESTED** [review]
Question about documenting both SPI1 **and SPI2**

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst (hunk 3 above)
Comment (rickconrey) [reply boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: I can change it if you'd like.  For the SOC, I added both SPI1 and SPI2 with all the different configurations in pinmux. For the nucleo_wb55rg dev board, I only enabled one configuration of SPI1, which is the one I documented.  Should I be enabling everything on a dev board as well?

**Review by tbursztyka — APPROVED** [review]

File: boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst (hunk 3 above)
Comment (dbkinder) [reply boards/arm/nucleo_wb55rg/doc/nucleowb55rg.rst]: If only SPI1 is enabled on the board, then the docs are just fine (I was thinking it was an oversight in the documentation).

**Review by dbkinder — APPROVED** [review]
+1 for docs

**Pull request merged**